var/spack/repos/builtin/packages/rocsolver/package.py

```diff
@@ -173,9 +175,13 @@ def check(self):
         "5.4.3",
         "5.5.0",
         "5.5.1",
+        "5.6.0",
+        "5.6.1",
     ]:
         depends_on("hip@" + ver, when="@" + ver)
         depends_on("rocblas@" + ver, when="@" + ver)
+    for ver in ["5.6.0", "5.6.1"]:
+        depends_on("rocsparse@" + ver, when="@" + ver)
```

Comment:
You don't need to enforce that the rocsparse version exactly matches the rocsolver version. This could just be `depends_on("rocsparse@5.2:", when="@5.6:")`. There's not very much sparse functionality in rocsolver, so updates that require newer versions of rocsparse will be rare.

Although, I might suggest applying https://github.com/ROCmSoftwarePlatform/rocSOLVER/commit/bab7bf71c2e7d1278d017e9ba087ef4aabebb940 as a patch to ROCm 5.6 and 5.7 to maximize source-compatibility with future versions of rocSPARSE.

Reply:
Thanks @cgmb, updated rocsolver recipe as per the comment.